app/controllers/time_slot_controller.rb

```diff
@@ -17,6 +17,12 @@ def schedule
         return
       end
 
+      if Appointment.where("patient_id = ? AND start > ?", current_patient.id, Time.zone.now).last != nil
+        Appointment.where("patient_id = ? AND start > ?", current_patient.id, Time.zone.now).find_each do |appointment|
+          appointment.destroy
+        end
+      end
```

Comment:
Aqui você consegue usar o próprio relacionamento do ActiveRecord pra filtrar os appointmentes do paciente. E também dá pra usar o `destroy_all` pra não precisar iterar com o `each` :smile: 
```suggestion
      current_patient.appointments.where('start > ?', Time.zone.now).destroy_all
```

Reply:
Realmente @GabrielaMafra, mais otimizada e enxuta sua sugestão, toda diminuição na complexidade da operação (por menor que seja) é bem vinda :grinning: